components/compositing/compositor.rs

```diff
@@ -840,6 +836,10 @@ impl IOCompositor {
                         },
                     }
                 }
+                // if compositor is animating, possibly due to image animation, we need to generate new frame.
+                if self.animations_or_animation_callbacks_running() {
+                    self.generate_frame(&mut txn, RenderReasons::RESOURCE_UPDATE);
+                }
```

Comment:
Hrm. This isn't the place to do this, I think. We should not be triggered a new frame every time images are updated. The script/layout that is getting the new images should know when its contents have changed and ask the compositor to repaint. Doing it automatically like this is going to lead to a lot of unnecessary rendering, I think.